compiler/modules/nback/structure.bal

```diff
@@ -352,30 +352,48 @@ function mappingOrderFields(t:MappingAtomicType mat, string[] fieldNames, bir:Op
 function buildMappingGet(llvm:Builder builder, Scaffold scaffold, bir:MappingGetInsn insn) returns BuildError? {
     bir:Register mappingReg = insn.operands[0];
     bir:StringOperand keyOperand = insn.operands[1];
-    int? fieldIndex = mappingFieldIndex(scaffold.typeContext(), mappingReg.semType, keyOperand);
-    boolean fill;
-    RuntimeFunction rf;
-    llvm:Value k;
-    if fieldIndex == () {
-        fill = insn.name == bir:INSN_MAPPING_FILLING_GET;
-        rf = fill ? mappingFillingGetFunction : mappingGetFunction;
-        k = check buildString(builder, scaffold, keyOperand);
+    var fieldIndexRes = mappingFieldIndex(scaffold.typeContext(), mappingReg.semType, keyOperand);
+    llvm:Value mapping = builder.load(scaffold.address(mappingReg));
+    if fieldIndexRes == () {
+        boolean fill = insn.name == bir:INSN_MAPPING_FILLING_GET;
+        RuntimeFunction rf = fill ? mappingFillingGetFunction : mappingGetFunction;
+        llvm:Value memberWithErr = <llvm:Value>builder.call(scaffold.getRuntimeFunctionDecl(rf), [mapping, check buildString(builder, scaffold, keyOperand)]);
+        llvm:Value member = fill ? buildCheckPanicCode(builder, scaffold, memberWithErr, insn.pos) : memberWithErr;
+        buildStoreMappingGet(builder, scaffold, insn, member, mapping, mappingReg, keyOperand);
```

Comment:
I think it will be simpler to say
```ballerina
if insn.name == bir:INSN_MAPPING_FILLING_GET {
    ...
}
else {
    buildNoPanicStoreMappingGet(...);
}
```
Please give a try and see.

Reply:
Maybe even combine this else case with else case at the end.

Reply:
> I think it will be simpler to say
> 
> ```ballerina
> if insn.name == bir:INSN_MAPPING_FILLING_GET {
>     ...
> }
> else {
>     buildNoPanicStoreMappingGet(...);
> }
> ```
> 
> Please give a try and see.

Changed

Reply:
> Maybe even combine this else case with else case at the end.

I think we cannot do that because llvm:Value for key is different